virtual_rainforest/main.py

```diff
@@ -0,0 +1,126 @@
+    should be handled appropriately downstream.
+
+    Args:
+        model_list: A list of models to select
+
+    Returns:
+        modules: A set of models to be configured
+    """
+
+    # Remove "core" from model list as it is not a model
+    if "core" in model_list:
+        model_list.remove("core")
+
+    LOGGER.info(f"Attempting to configure the following models: {model_list}")
+
+    # Make list of missing models, and return an error if necessary
+    miss_model = [model for model in model_list if model not in MODEL_REGISTRY.keys()]
+    if miss_model != []:
+        LOGGER.error(
+            f"The following models cannot be configured as they are not found in the "
+            f"registry: {miss_model}"
+        )
+        return None
+
+    # Then look for each model in the registry
+    modules = [MODEL_REGISTRY[model] for model in model_list]
+
+    return modules
+
+
+def configure_models(
+    config: dict[str, Any], model_list: list[Type[BaseModel]]
+) -> list[BaseModel]:
+    """Configure a set of models for use in a `virtual_rainforest` simulation.
+
+    Args:
+        config: The full virtual rainforest configuration
+        modules: A set of models to be configured
+
+    Returns:
+        models_cfd: A set of configured models
+    """
+
+    # Use factory methods to configure the desired models
+    models_cfd = [model.factory(config) for model in model_list]
+    return models_cfd
+
+
+def vr_run(
+    cfg_paths: Union[str, list[str]], output_folder: str, out_file_name: str
+) -> None:
+    """Perform a virtual rainforest simulation.
+
+    This is a high-level function that runs a virtual rainforest simulation. At the
+    moment this involves validating an input configuration, and using this configuration
+    to generate a set of configured model objects suitable for downstream use. Down the
+    line this should be extended to encompass far more steps.
+
+    Args:
+        cfg_paths: Set of paths to configuration files
+        output_folder: Folder to save combined configuration to
+        out_file_name: Name for the combined configuration file
+    """
+
+    config = validate_config(cfg_paths, output_folder, out_file_name)
+
+    model_list = select_models(deepcopy(config["core"]["modules"]))
+
+    if model_list is None:
+        log_and_raise(
+            "Could not find all the desired models, ending the simulation.",
+            InitialisationError,
+        )
+        return
+    else:
+        LOGGER.info(
+            "All models found in the registry, now attempting to configure them."
+        )
+
+    models_cfd = configure_models(config, model_list)
+
+    if any(model is None for model in models_cfd):
+        log_and_raise(
+            "Could not configure all the desired models, ending the simulation.",
+            InitialisationError,
+        )
+        return
+    else:
+        LOGGER.info(
+            "All models successfully configured, now attempting to initialise them."
+        )
```

Comment:
Similarly, why is this not in `configure models` (which is otherwise a one liner).